File: frontend/src/types/feedback.ts
export type FeedbackStatus = 'pending' | 'processing' | 'resolved'

export interface FeedbackPublic {
  id: number
  subject: string
  content: string
  status: FeedbackStatus | string
  admin_reply: string | null
  created_at: string
  updated_at: string
}

export interface FeedbackCreate {
  subject: string
  content: string
}

export interface FeedbackAdminItem extends FeedbackPublic {
  user_id: number
  username: string
  nickname: string
}

export interface FeedbackAdminUpdate {
  status: FeedbackStatus
  admin_reply?: string
}

File: backend/app/schemas/feedback.py
from __future__ import annotations

from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, Field

FeedbackStatus = Literal["pending", "processing", "resolved"]


class FeedbackCreate(BaseModel):
    subject: str = Field(min_length=1, max_length=120)
    content: str = Field(min_length=1, max_length=5000)


class FeedbackPublic(BaseModel):
    model_config = {"from_attributes": True}

    id: int
    subject: str
    content: str
    status: str
    admin_reply: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class FeedbackAdminItem(FeedbackPublic):
    user_id: int
    username: str
    nickname: str


class FeedbackAdminUpdate(BaseModel):
    status: FeedbackStatus
    admin_reply: Optional[str] = Field(None, max_length=2000)
